File: src/main/java/tn/esprit/models/Activities.java
package tn.esprit.models;

import java.util.List;

public class Activities {

    //id,name,img,state,price,description
    private int id;
    private String name, img, state, description;
    private float price;
    private List<Events> events;

    public Activities() {
    }

    public Activities(String name, String img, String state, String description, float price) {
        this.name = name;
        this.img = img;
        this.state = state;
        this.description = description;
        this.price = price;
    }

    public Activities(int id, String name, String img, String state, String description, float price) {
        this.id = id;
        this.name = name;
        this.img = img;
        this.state = state;
        this.description = description;
        this.price = price;
    }

    public Activities(int i, int i1, String name, String image, String california, String testDescription, int i2) {
    }


    public Activities(String activityName) {
        this.name=activityName;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public float getPrice() {
        return price;
    }

    public void setPrice(float price) {
        this.price = price;
    }

    public List<Events> getEvents() {
        return events;
    }

    public void setEvents(List<Events> events) {
        this.events = events;
    }

    @Override
    public String toString() {
        return "Activities{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", img='" + img + '\'' +
                ", state='" + state + '\'' +
                ", description='" + description + '\'' +
                ", price=" + price +
                '}';
    }
}

File: src/Entity/Activities.php
<?php

namespace App\Entity;

use App\Repository\ActivitiesRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ActivitiesRepository::class)]
class Activities
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    
    private ?string $name = null;

    #[ORM\Column(length: 255)]
    
    private ?string $img = null;

    #[ORM\Column(length: 255)]
    
    private ?string $state = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: 2)]
   
    private ?string $price = null;

    #[ORM\Column(length: 255)]
    
    private ?string $description = null;

    #[ORM\ManyToOne(inversedBy: 'Activities')]
    private ?Events $events = null;

    #[ORM\ManyToMany(targetEntity: Events::class, mappedBy: 'events')]
    private Collection $event;

    public function __construct()
    {
        $this->event = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): static
    {
        $this->name = $name;

        return $this;
    }

    public function getImg(): ?string
    {
        return $this->img;
    }

    public function setImg(string $img): static
    {
        $this->img = $img;

        return $this;
    }

    public function getState(): ?string
    {
        return $this->state;
    }

    public function setState(string $state): static
    {
        $this->state = $state;

        return $this;
    }

    public function getPrice(): ?string
    {
        return $this->price;
    }

    public function setPrice(string $price): static
    {
        $this->price = $price;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function getEvents(): ?Events
    {
        return $this->events;
    }

    public function setEvents(?Events $events): static
    {
        $this->events = $events;

        return $this;
    }
    public function __toString(): string
    {
        return $this->getName(); // Assuming 'name' is a property of the Activities entity
    }

    /**
     * @return Collection<int, Events>
     */
    public function getEvent(): Collection
    {
        return $this->event;
    }

    public function addEvent(Events $event): static
    {
        if (!$this->event->contains($event)) {
            $this->event->add($event);
            $event->addEvent($this);
        }

        return $this;
    }

    public function removeEvent(Events $event): static
    {
        if ($this->event->removeElement($event)) {
            $event->removeEvent($this);
        }

        return $this;
    }
   
}
